broker-core/src/main/java/org/apache/qpid/server/transport/MultiVersionProtocolEngine.java

```diff
@@ -464,13 +464,16 @@ public void received(QpidByteBuffer msg)
 
                     _broker.getEventLogger().message(new PortLogSubject(_port),
                                                      PortMessages.UNSUPPORTED_PROTOCOL_HEADER(Functions.str(headerBytes),
-                                                                                              supportedReplyVersion.toString()));
+                                                                                              String.valueOf(supportedReplyVersion)));
 
-                    try (QpidByteBuffer supportedReplyBuf = QpidByteBuffer.allocateDirect(supportedReplyBytes.length))
+                    if (supportedReplyBytes != null)
```

Comment:
The broker is required by the AMQP specification to send a reply with a latest supported AMQP version.

Thus, if supportedReplyBytes is null, it is a defect.

I would like to suggest  adding a check whether supportedReplyBytes is null. If it is null, we we can iterate via available creators and pickup the highest version among available and set supportedReplyBytes for the highest available protocol.

if there is no creator, that would mean that all protocols are disabled.... In that case the broker cannot support messaging... I am not sure what the broker behaviour should be in this case... It seems that broker is inconsistent state and possibly would need to bring down...